frontend/rolecall/src/app/piece/piece_editor.component.ts

```diff
@@ -88,22 +91,37 @@ export class PieceEditor implements OnInit {
   }
 
   private buildRenderingList() {
-    this.renderingItems = this.workingPieces.map(
-        (workingPiece, workingPieceIndex) => {
-      const hasNoChidren = workingPiece.type === 'SEGMENT'
-          ? false : workingPiece.positions.length === 0;
-      const sortPrefix = hasNoChidren ? ' ' : 'z';
-      const name = hasNoChidren ? '*' + workingPiece.name : workingPiece.name;
+    // Remove Super Ballet children
```

Comment:
All the new logic that's building `displayedPieces` looks like it can be extracted to a separate function.

Reply:
Done.